### PR TITLE
Clarify how the bash shell will be invoked if not specified explicitlly

### LINKED ISSUE #19191: Clarify how the bash shell will be invoked if not specified explicitly for jobs.<job_id>.steps[*].shell
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

### What part(s) of the article would you like to see updated?

Description for:
image
should be extended to say that the `bash -e {0}` will be applied if the shell parameter is not specified explicitly.

### Additional information

This change is related to the [comment](https://github.com/actions/runner/issues/1955#issuecomment-1189337732) in the runner repo.

### PR DESCRIPTION
**Pull request by nikola-jokic**
### Why:
This change is related to the https://github.com/actions/runner/issues/1955#issuecomment-1189337732 in the runner repo.
Closes #19191 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Added one sentence stating that, if not specified explicitly, bash will be invoked with `bash -e {0}`

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: content/actions/using-workflows/workflow-syntax-for-github-actions.md [hunk 1]
```diff
@@ -538,7 +538,7 @@ You can override the default shell settings in the runner's operating system usi
 
 | Supported platform | `shell` parameter | Description | Command run internally |
 |--------------------|-------------------|-------------|------------------------|
-| All | `bash` | The default shell on non-Windows platforms with a fallback to `sh`. When specifying a bash shell on Windows, the bash shell included with Git for Windows is used. | `bash --noprofile --norc -eo pipefail {0}` |
+| All | `bash` | The default shell on non-Windows platforms with a fallback to `sh`. Bash will be invoked with `bash -e {0}` if the shell parameter is not specified explicitly. When specifying a bash shell on Windows, the bash shell included with Git for Windows is used. | `bash --noprofile --norc -eo pipefail {0}` |
```
[the remark this answers is not in the document]
Comment (mattpollard) [reply content/actions/using-workflows/workflow-syntax-for-github-actions.md:541]: @eric-wieser's suggestion looks good 👍🏻 Should we still include @nikola-jokic's motion of `bash -e {0}`?

One thing I'd consider tweaking is this:

> Note that the behavior is different to specifying `bash` explicitly.

What's "the behavior" here? Can we be more explicit? "The behavior when not specifying `bash` in the path", etc.

**Review by mattpollard — CHANGES_REQUESTED** [review]
@nikola-jokic 👋🏻 If you could incorporate any changes from @eric-wieser and my comment, we'd greatly appreciate it 🙏🏻 Let us know when this is ready for a second look and we'll work with you to get it published.

File: content/actions/using-workflows/workflow-syntax-for-github-actions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (eric-wieser) [reply content/actions/using-workflows/workflow-syntax-for-github-actions.md:541]: "the behavior" is "the behavior described by this row of the table", which seems already clear to me.

[the remark this answers is not in the document]
Comment (eric-wieser) [reply content/actions/using-workflows/workflow-syntax-for-github-actions.md:541]: Perhaps reordering the sentences would make it clearer?
```suggestion
| Linux / macOS | unspecified | The default shell on non-Windows platforms. Note that runs a different command to when `bash` is specified explicitly. If `bash` is not found in the path, this is treated as `sh`. | `bash -e {0}` |
| All | `bash` | The default shell on non-Windows platforms with a fallback to `sh`. When specifying a bash shell on Windows, the bash shell included with Git for Windows is used. | `bash --noprofile --norc -eo pipefail {0}` |
```

[the remark this answers is not in the document]
Comment (nikola-jokic) [reply content/actions/using-workflows/workflow-syntax-for-github-actions.md]: @eric-wieser great suggestion! Thank you!

**Comment by nikola-jokic** [comment]
Thank you so much @mattpollard for the review and @eric-wieser for suggestions! I applied the latest suggestion, and it seems clear to me. Since the flags are changing the way the action behaves (`--pipefail` for example is fail-fast behaviour), I would leave it as is. Not sure how to be more explicit there but if you want to change anything, I am happy to apply your changes, or you can commit it right away :relaxed:

**Review by mattpollard — APPROVED** [review]
Thanks for being open to feedback, @nikola-jokic! I'll merge this.

**Comment by tarek29577** [comment]
Stop ✋

في الأربعاء، ١٠ أغسطس ٢٠٢٢ ١٢:٥١ م github-actions[bot] <
***@***.***> كتب:

**Pull request merged**